client/webserver/site/src/js/forms.ts

```diff
@@ -698,7 +699,8 @@ export class WalletConfigForm {
   reorder (box: HTMLElement) {
     const inputs: Record<string, HTMLElement[]> = {}
     box.querySelectorAll('input').forEach((input: ConfigOptionInput) => {
-      const k = input.configOpt.key
+      const k = input.dataset.configKey
+      if (!k) return // typescript
```

Comment:
What does this comment indicate?

Reply:
That I am doing this check because of typescript, can remove if you'd like

Reply:
Docs are always good, I'm just still confused.  You mean this is something typescript-specific and not javascript?  
What does it indicate if `!k`?

Reply:
that's correct, it is typescript-specific and if not added it will throw the following error:

```
src/js/forms.ts:705:14 - error TS2538: Type 'undefined' cannot be used as an index type.

705       inputs[k] = els
```

It happens because undefined is a valid value for `input.dataset.configkey`

Reply:
I can also be more specific at the comment

Reply:
When and why would it be undefined?

Reply:
It is a valid value, it does not mean it will be. It shouldn't, that's why the comment

Reply:
Ohhh, it's just `tsc` code analysis being obnoxiou.  It's not that typescript and javascript syntax are different, or even that babel will fail to transpile this to JavaScript without this check, but that we hit `TS2538`.

Reply:
If you change this comment to `// TS2538` we can merge it.